styx-scheduler-service/src/main/java/com/spotify/styx/state/QueuedStateManager.java

```diff
@@ -288,8 +298,16 @@ private void process(RunState state, long counterPosition, Event event,
       throw t;
     }
 
+    // If the service is stopped right before this line, the event was successfully stored but never
+    // going to be consumed. Event consumption is only best effort with the current implementation.
```

Comment:
this is true for output handlers as well, right?

Reply:
It is true. Are you suggesting to remove the comment or adding more details to the existing one?

Reply:
not sure, but i find it a bit misleading as it is right now

Reply:
I am ok with removing this comment altogether and if needed explain more in-depth what these racing conditions might cause and what guarantees they break in the docs, in a separate PR.

Reply:
sounds good